### PR TITLE
feat(internal): repository cache revision

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Adds a cache revision number which is part of validation.

## Context:

This gives us a lever to pull to invalidate existing caches whenever we make breaking internal changes to scan or branch cache.

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/cache/repository/index.spec.ts [hunk 1]
```diff
@@ -38,28 +38,16 @@ describe('lib/util/cache/repository', () => {
...
-    fs.readFile.mockResolvedValueOnce('{"repository":"abc/def"}' as any);
-    await repositoryCache.initialize({
-      ...config,
-      repositoryCache: 'enabled',
-    });
-    await repositoryCache.finalize();
-    expect(fs.readFile.mock.calls).toHaveLength(1);
-    expect(fs.outputFile.mock.calls).toHaveLength(1);
-  });
-  it('migrates', async () => {
```
Comment (rarkins) [inline lib/util/cache/repository/index.spec.ts:50]: Caches will be invalidated whenever the revision changes, so any existing migrations aren't necessary.

**Comment by zharinov** [comment]
For old revisions data, should it be evicted explicitly or it will happen implicitly at some moment?

**Review by viceice — APPROVED** [review]

File: lib/util/cache/repository/index.ts [hunk 2]
```diff
@@ -56,7 +60,11 @@ export function getCacheFileName(config: RenovateConfig): string {
 }
 
 function validate(config: RenovateConfig, input: any): Cache | null {
-  if (input?.repository === config.repository) {
+  if (
+    input &&
+    input.repository === config.repository &&
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/util/cache/repository/index.ts:65]: Readability

**Comment by rarkins** [comment]
> For old revisions data, should it be evicted explicitly or it will happen implicitly at some moment?

It will happen implicitly because the old data will be overwritten during finalise

**Pull request merged**